src/renderer/Renderer.js

```diff
@@ -217,7 +133,8 @@ export default class Renderer {
 
         this._updateDataframeMatrices(dataframes);
 
-        this._runViewportAggregations(renderLayer);
+        renderLayer.parseVizExpression = parseVizExpression; // Important! to avoid a circular dependency problem (eg. viewportFeatures)
```

Comment:
👍

Reply:
@rochoa Here it is a diagram of this situation:

![carto vl - circular dependency 1](https://user-images.githubusercontent.com/458196/49230559-44657300-f3f0-11e8-81bf-ba23d6e3f692.png)

As you can see there are 2 places where this problem manifests itself, both related to blendTo  using strings.

This PR is around the 'Point 2', where the work-around is passing the reference to the function... several classes down to its final use. The approach is "similar" to the one used in "Point 1" by David at https://github.com/CartoDB/carto-vl/commit/80453d8f9c44bd8afe4b9630175b61882f9b5d87

It would be great to break this cycle, changing the way `parser` is currently using the expressions (as the parsing from the expressions seems interesting)

Reply:
After our conversation, I'm gonna merge this one and create a new issue to tackle this in the future

Reply:
Cool, thanks. And thank YOU for the diagram. 👏